ls1intum/Artemis
# [Bugfix] Fix the error handling for assessments of prog. ex. before a due date

### Checklist

### Motivation and Context


An incomprehensible error is thrown when a tutor opens the exercise assessment page of a programming exercise whose 'build after due date' is not reached.

### Description

The server was throwing a forbidden() when `checkIfExerciseDueDateIsReached` returned false. Now a `AccessForbiddenException` is thrown with a more explaining info message.

### Steps for Testing


1. Log in to Artemis
2. Create a programming exercise with a `build after due date`
3. Log in as a student and participate
4. Log in as a tutor and go to the Assessment Dashboard of the course
5. Click on Exercise Dashboard of the programming exercise
6. Verify that the error message on top is displayed correctly

### Test Coverage



Stays the same as no new functionality is added.

### Screenshots


Course Assessment Dashboard:
![image](https://user-images.githubusercontent.com/33484318/106113224-857f7c00-614e-11eb-9c8c-b30fb1c2e0b6.png)

Exercise Assessment Dashboard:
![image](https://user-images.githubusercontent.com/33484318/106113361-b19afd00-614e-11eb-911b-5b638f527215.png)
